### PR TITLE
Add support for KeyVault credentials

### PR DESCRIPTION
**Pull request by kpajdzik**

### CODE REVIEW

File: lib/credentials/keyVaultCredentials.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+    }
+  }
+
+  signRequest(webResource: WebResource): Promise<WebResource> {
+    // authentication is provided by the createSigningFilter method.
+    return Promise.resolve(undefined);
+  }
+
+  createSigningFilter(): (resource: WebResource, next: Function, callback: ServiceCallback<any>) => any {
+    const self = this;
```
Comment (daschult) [inline lib/credentials/keyVaultCredentials.ts]: Is this needed for TypeScript? I thought `this` could be used without creating `self` variables.

File: lib/credentials/keyVaultCredentials.ts [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+    }
+
+    if (!this.authenticator) {
+      this.authenticator = this._authenticatorMapper(this.credentials);
+    }
+  }
+
+  signRequest(webResource: WebResource): Promise<WebResource> {
+    // authentication is provided by the createSigningFilter method.
+    return Promise.resolve(undefined);
```
Comment (RikkiGibson) [inline lib/credentials/keyVaultCredentials.ts]: Does it make sense to implement ServiceClientCredentials if this is just a dummy method? It doesn't make sense to ever call this `signRequest`, does it?

File: lib/credentials/keyVaultCredentials.ts (hunk 1 above)
Comment (kpajdzik) [reply lib/credentials/keyVaultCredentials.ts]: Yes, but we need to use => notation, right? I didn't want to change too much in the code.

Comment (RikkiGibson) [reply lib/credentials/keyVaultCredentials.ts]: Riight, once you're in the `function() {}` scope, `this` isn't from the KeyVaultCredentials any more.

File: lib/credentials/keyVaultCredentials.ts (hunk 2 above)
Comment (kpajdzik) [reply lib/credentials/keyVaultCredentials.ts]: That's true. It's mirrored from Node SDK and I think it's worth keeping the object structure. What do you think?

Comment (kpajdzik) [reply lib/credentials/keyVaultCredentials.ts]: I've created an issue for that: https://github.com/Azure/ms-rest-nodeauth/issues/20

**Pull request merged**